From a1130234b08dee69ce6623b4d0373584c179b55d Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 28 Jun 2024 09:13:34 +0300
Subject: [PATCH] [FSW-420] fs1: initialize security efuse sys cfg

with this patch, initialized efuse sys cfg and
sys cfg2 dwords in security data structure with
efuse sys configuration in boot rom global.
all security related functions can use the same
efuse dwords under sec data structure and used
to export as boot info.

Change-Id: I2ecf9a11095681e05073ff517910a37c9aa88bc4
---

diff --git a/common/security.c b/common/security.c
index 47ff6c3..912b05e 100644
--- a/common/security.c
+++ b/common/security.c
@@ -5,14 +5,27 @@
  *
  */
 
+#include <boot_rom.h>
+
 static void security_handle_jtag_if(void)
 {
 	/* TODO: add jtag enable/disable logic */
+}
+
+static void security_load_sec_efuse_dword(void)
+{
+	boot_rom_global.sec_data.efuse_sys_cfg = boot_rom_global.config_fuse;
+	boot_rom_global.sec_data.efuse_sys_cfg2 = boot_rom_global.config2_fuse;
 }
 
 void boot_rom_security_init(void)
 {
 	/* Based on efuse enable/disable jtag interface and clear FSM timer */
 	security_handle_jtag_if();
+
+	if (boot_rom_global.die_type != IO_DIE_MASTER)
+		return;
+
+	security_load_sec_efuse_dword();
 }
 
diff --git a/include/boot_rom.h b/include/boot_rom.h
index ac80bde..dcf35e1 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -10,6 +10,7 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include "hl_boot_desc_rom.h"
+#include <hl_sec_if.h>
 
 #define wfe()	__asm__ __volatile__("sleep 0x11\n" : : )
 
@@ -90,6 +91,7 @@
 	bool fake_fuse_en;
 	uint32_t fake_fuse_spi_offset;
 	bool boot_info_set_done;
+	struct hl_fw_sec_info sec_data;
 };
 
 struct boot_rom_manifest_s {
